contracts/Joint.sol

```diff
@@ -292,369 +358,418 @@ abstract contract Joint is ySwapper {
+        _path = new address[](isReferenceToken || is_internal ? 2 : 3);
         _path[0] = _token_in;
-        if (is_weth || is_internal) {
+        if (isReferenceToken || is_internal) {
             _path[1] = _token_out;
         } else {
-            _path[1] = address(WETH);
+            _path[1] = address(referenceToken);
             _path[2] = _token_out;
         }
     }
 
     function getReward() internal virtual;
 
-    function depositLP() internal virtual;
+    function depositLP() internal virtual {}
+
+    function withdrawLP() internal virtual {}
 
-    function withdrawLP() internal virtual;
+    struct tokenAmount {
+        address token;
+        uint256 amount;
+    }
 
-    function swapReward(uint256 _rewardBal)
+    /*
+     * @notice
+     *  Function available internally swapping amounts necessary to swap rewards
+     * @return tokenAmount array of the swap path followed
+     */
+    function swapRewardTokens()
         internal
         virtual
-        returns (address, uint256)
+        returns (tokenAmount[] memory)
     {
-        if (reward == tokenA || reward == tokenB || _rewardBal == 0) {
-            return (reward, 0);
-        }
-
-        if (tokenA == WETH || tokenB == WETH) {
-            return (WETH, sellCapital(reward, WETH, _rewardBal));
-        }
-
-        // Assume that position has already been liquidated
-        (uint256 ratioA, uint256 ratioB) =
-            getRatios(balanceOfA(), balanceOfB(), investedA, investedB);
-        if (ratioA >= ratioB) {
-            return (tokenB, sellCapital(reward, tokenB, _rewardBal));
+        tokenAmount[] memory _swapToAmounts = new tokenAmount[](
+            rewardTokens.length
+        );
+        for (uint256 i = 0; i < rewardTokens.length; i++) {
+            address reward = rewardTokens[i];
+            uint256 _rewardBal = IERC20(reward).balanceOf(address(this));
+            // If the reward token is either A or B, don't swap
+            if (reward == tokenA || reward == tokenB || _rewardBal == 0) {
+                _swapToAmounts[i] = tokenAmount(reward, 0);
+            // If the referenceToken is either A or B, swap rewards against it 
+            } else if (tokenA == referenceToken || tokenB == referenceToken) {
+                _swapToAmounts[i] = tokenAmount(
+                    referenceToken,
+                    swap(reward, referenceToken, _rewardBal)
+                );
+            } else {
+                // Assume that position has already been liquidated
+                (uint256 ratioA, uint256 ratioB) = getRatios(
+                    balanceOfA(),
+                    balanceOfB(),
+                    investedA,
+                    investedB
+                );
+                address swapTo = (ratioA >= ratioB) ? tokenB : tokenA;
+                _swapToAmounts[i] = tokenAmount(
+                    swapTo,
+                    swap(reward, swapTo, _rewardBal)
+                );
```

Comment:
linking with previous comments about `swapRewardTokens` returning `totalAmountOutA, totalAmountOutB`

replace these lines with: 
```
if(swapTo == tokenA) {
   totalAmountOutA += swap(reward, swapTo, _rewardBal);
}
```

and add the variables to the returned variables of the function

Reply:
Resolving as implemented in comment above